python/revng/pypeline/object.py

```diff
@@ -225,6 +230,17 @@ def deserialize(cls, obj: str) -> ObjectID:
         """Convert a string representation back to an object"""
         raise NotImplementedError()
 
+    @abstractmethod
+    def serialize_bytes(self) -> bytes:
+        """Convert the object to a bytes representation that can be later be deserialized"""
+        raise NotImplementedError()
```

Comment:
This is an understatement.

We need to put a bigass explanation about the expectations we have.
If you don't want to specify the full format (byte discriminator + data + byte discriminator + ...), we have at very least to explain the fact that these need to be lexicographically sortable, and explaining what we mean by that.

Reply:
Changed the docstring to document the expected the binary format